src/admin/trades.ts

```diff
@@ -69,24 +76,79 @@ var TradesListController = ($scope : TradesScope, $log : ng.ILogService, subscri
+          $scope.gridApi = gridApi;
+        }
     };
 
-    var addTrade = t => $scope.trade_statuses.push(new DisplayTrade(t));
+    var addTrade = t => {
+      if (t.alloc<0) {
+        for(var i = 0;i<$scope.trade_statuses.length;i++) {
+          if ($scope.trade_statuses[i].tradeId==t.tradeId) {
+            $scope.trade_statuses.splice(i, 1);
+            break;
+          }
+        }
+      } else {
+        var exists = false;
+        for(var i = 0;i<$scope.trade_statuses.length;i++) {
+          if ($scope.trade_statuses[i].tradeId==t.tradeId) {
+            exists = true;
+            $scope.trade_statuses[i].time = t.time;
+            $scope.trade_statuses[i].quantity = t.quantity;
+            $scope.trade_statuses[i].value = t.value;
+            $scope.trade_statuses[i].alloc = t.alloc;
+            $scope.trade_statuses[i].allocprice = t.allocprice;
+            if ($scope.trade_statuses[i].alloc >= $scope.trade_statuses[i].quantity)
+              $scope.trade_statuses[i].side = 'K';
+            if ($scope.audioReady && $scope.audio) {
+                var audio = new Audio('/audio/erang.mp3');
+                audio.volume = 0.5;
+                audio.play();
+            }
+            break;
+          }
+        }
+        if (!exists) {
+          $scope.trade_statuses.push(new DisplayTrade($scope, t));
+          if ($scope.audioReady && $scope.audio) {
+              var audio = new Audio('/audio/boom.mp3');
+              audio.volume = 0.5;
+              audio.play();
+          }
+        }
+      }
+    };
 
     var sub = subscriberFactory.getSubscriber($scope, Messaging.Topics.Trades)
         .registerConnectHandler(() => $scope.trade_statuses.length = 0)
         .registerDisconnectedHandler(() => $scope.trade_statuses.length = 0)
         .registerSubscriber(addTrade, trades => trades.forEach(addTrade));
 
+    var updateQP = qp => {
+      $scope.audio = qp.audio;
+      $scope.gridOptions.columnDefs[$scope.gridOptions.columnDefs.map(function (e) { return e.field; }).indexOf('alloc')].visible = (qp.mode === Models.QuotingMode.Boomerang);
+      $scope.gridOptions.columnDefs[$scope.gridOptions.columnDefs.map(function (e) { return e.field; }).indexOf('allocprice')].visible = (qp.mode === Models.QuotingMode.Boomerang);
+      $scope.gridApi.grid.refresh();
+    };
+
+    var qpSub = subscriberFactory.getSubscriber($scope, Messaging.Topics.QuotingParametersChange)
+        .registerConnectHandler(() => $scope.trade_statuses.length = 0)
+        .registerDisconnectedHandler(() => $scope.trade_statuses.length = 0)
+        .registerSubscriber(updateQP, qp => qp.forEach(updateQP));
+
     $scope.$on('$destroy', () => {
         sub.disconnect();
-        $log.info("destroy trades list");
+        qpSub.disconnect();
+        // $log.info("destroy trades list");
     });
 
-    $log.info("started trades list");
+    // $log.info("started trades list");
+    setTimeout(function(){$scope.audioReady = true;},7000);
```

Comment:
One idea is to affix a is-loaded-from-the-database-on-startup flag to all the Trade objects, then not triggering the sound on those trades. I could see a flag like that being useful in other contexts.